assets/src/edit-story/components/library/panes/pageTemplates/savedTemplates.js

```diff
@@ -61,7 +71,35 @@ function SavedTemplates({ pageSize, setShowDefaultTemplates }) {
     }
   }, [loadTemplates, pageTemplates]);
 
-  // @todo Saving template should belong to the virtual list, it's currently misplaced.
+  const onClickDelete = useCallback(({ postId }, e) => {
+    e?.stopPropagation();
+    if (postId) {
+      setShowDialog(true);
+      setTemplateToDelete(postId);
+    }
+  }, []);
+  const handleDelete = useCallback(
+    () =>
+      deletePageTemplate(templateToDelete)
+        .then(() => {
+          setPageTemplates(
+            pageTemplates.filter(({ postId }) => postId !== templateToDelete)
+          );
+          setShowDialog(false);
+        })
```

Comment:
Hm, where is the state kept currently? Only locally here? Are we reloading all the templates every time we switch to this panel? And if not, isn't the deleted template reappearing if we switch out of this panel and back again?

We should keep this state somewhere locally permanently (and probably not in the API provider).

Reply:
> Are we reloading all the templates every time we switch to this panel?

Yes, currently the templates are reloaded. Are you thinking we should only get the update templates only once? I wonder if this could cause unexpected results if multiple users are adding/removing templates at the same time.